tippicserver/views_public.py

```diff
@@ -268,36 +269,56 @@ def get_transactions_api():
 
     each item in the list contains:
         - the tx_hash
-        - tx direction (in, out)
         - amount of kins transferred
         - date
         - title and additional details
     """
+    import urllib.request, json
+    from tippicserver.models.user import get_address_by_userid
     detailed_txs = []
     try:
         user_id, auth_token = extract_headers(request)
-        server_txs = [{'type': 'server', 'tx_hash': tx.tx_hash, 'amount': tx.amount, 'client_received': not tx.incoming_tx, 'tx_info': tx.tx_info, 'date': arrow.get(tx.update_at).timestamp} for tx in list_user_transactions(user_id, MAX_TXS_PER_USER)]
-
-        # # get the offer, task details
-        # for tx in server_txs:
-        #     details = get_offer_details(tx['tx_info']['offer_id']) if not tx['client_received'] else get_task_details(tx['tx_info']['task_id'])
-        #     detailed_txs.append({**tx, **details})
-
-        # get p2p details
-        import emoji
-        kin_from_a_friend_text=emoji.emojize(':party_popper: Kin from a friend')
-        p2p_txs = [{'title': kin_from_a_friend_text if str(tx.receiver_user_id).lower() == str(user_id).lower() else 'Kin to a friend',
-                    'description': 'a friend sent you %sKIN' % tx.amount,
-                    'provider': {'image_url': 'https://s3.amazonaws.com/kinapp-static/brand_img/poll_logo_kin.png', 'name': 'friend'},
-                    'type': 'p2p',
-                    'tx_hash': tx.tx_hash,
-                    'amount': tx.amount,
-                    'client_received': str(tx.receiver_user_id).lower() == str(user_id).lower(),
-                    'tx_info': {'memo': 'na', 'task_id': '-1'},
-                    'date': arrow.get(tx.update_at).timestamp} for tx in list_p2p_transactions_for_user_id(user_id, MAX_TXS_PER_USER)]
-
-        # merge txs:
-        detailed_txs = detailed_txs + p2p_txs
+        public_address = get_address_by_userid(user_id)
+        platform = get_user_os_type(user_id)
+        link = DISCOVERY_APPS_ANDROID_URL if platform == OS_ANDROID else DISCOVERY_APPS_OSX_URL
+        with urllib.request.urlopen(link) as url:
+            discover_apps = json.loads(url.read().decode())['apps']
+            
+            for tx in list_user_transactions(user_id, MAX_TXS_PER_USER):    
+                if tx.tx_type == APP_TO_APP:
+                    app_data = next(item for item in discover_apps if item['memo'] == tx.tx_for_item_id)
```

Comment:
we need to change the naming on the json
instead of  'memo' we should use 'app_id' or 'kin_app_id'
This needs to be changed on json, move-kin-android, and this code here. 
Let's do this together though so that we don't get out of sync...